migrator/README.md

```diff
@@ -1,23 +1,98 @@
 # StackRox Database migration
 
+## Purpose of database migrations
+
+When changing the data model for stored objects, data conversions may be required. All migrations
+are in the `migrations` subdirectory.
+
+Migrations are organized with sequence numbers and executed in sequence. Each migration is provided
+with a pointer to `types.Databases` where the pointed object contains all necessary database instances,
+including `Bolt`, `RocksDB`, `Postgres` and `GormDB` (datamodel for postgres).
+
+A migration can read from any of the databases, make changes to the data or to the datamodel
+(database schema when working with postgres), then persist these changes to the database.
+
 ## How to write new migration script
 
 Script should correspond to single change. Script should be part of the same release as this change.
 Here are the steps to write migration script:
 
-1. Lookup current database version in `pkg/migrations/internal/seq_num.go` file
-2. Under `migrations` folder create new folder with name
-`m_{currentDBVersion}_to_m_{currentDBVersion+1}_{summary_of_migration}`
-    1. Ensure that the `summary_of_migration` follows the naming convention of previous migrations, i.e., postfix `_policy` if it modifies policies
-3. Create at least two files: `migration.go` and `migration_test.go`. These files should belong to package `m{currentDBVersion}tom{currentDBVersion+1}`
-4. To better understand how to write these two files, look at existing examples: [#1](https://github.com/stackrox/rox/pull/8609) [#2](https://github.com/stackrox/rox/pull/7581) [#3](https://github.com/stackrox/rox/pull/7921) in `migrations` directory. Avoid depending on code that might change in the future as **migration should produce consistent results**.
-5. Add to `migrator/runner/all.go` line
+1. Determine which database group will be targeted. The options should be `BoltDB or RocksDB`,
+`Postgres and Gorm` or a migration from the former to the latter.
+
+    Note: the rule of thumb to determine the target group is based on the first release where the migration should be
+    applied.
+    1. Before 3.73, the only target group should be `BoltDB or RocksDB`.
```

Comment:
We are already in 74. We should not recommend to add new migration on top of legacy databases.